pkg/synopsysctl/cmd_migrate.go

```diff
@@ -27,59 +27,374 @@ import (
+		ns.Labels = util.InitLabels(ns.Labels)
+		ns.Labels[fmt.Sprintf("synopsys.com/%s.%s", appName, name)] = version
+		_, err = util.UpdateNamespace(kubeClient, ns)
+		if err != nil {
+			return fmt.Errorf("error updating %s namespace due to %+v", namespace, err)
+		}
+	}
+	return nil
+}
+
+// addNameLabels adds a name label to all its resources
+func addNameLabels(namespace string, name string, appName string) error {
+	deployments, err := util.ListDeployments(kubeClient, namespace, fmt.Sprintf("app=%s", appName))
+	if err != nil {
+		return fmt.Errorf("unable to list deployments for %s %s in namespace %s due to %+v", appName, name, namespace, err)
+	}
+
+	for _, deployment := range deployments.Items {
+		if _, ok := deployment.Labels["name"]; !ok {
+			deployment.Labels = util.InitLabels(deployment.Labels)
+			deployment.Labels["name"] = name
+			deployment.Spec.Template.Labels = util.InitLabels(deployment.Spec.Template.Labels)
+			deployment.Spec.Template.Labels["name"] = name
+			_, err = util.UpdateDeployment(kubeClient, namespace, &deployment)
+			if err != nil {
+				return fmt.Errorf("unable to update %s deployment in namespace %s due to %+v", deployment.GetName(), namespace, err)
+			}
+		}
+	}
+
+	rcs, err := util.ListReplicationControllers(kubeClient, namespace, fmt.Sprintf("app=%s", appName))
+	if err != nil {
+		return fmt.Errorf("unable to list replication controllers for %s %s in namespace %s due to %+v", appName, name, namespace, err)
+	}
+
+	for _, rc := range rcs.Items {
+		if _, ok := rc.Labels["name"]; !ok {
+			rc.Labels = util.InitLabels(rc.Labels)
+			rc.Labels["name"] = name
+			rc.Spec.Template.Labels = util.InitLabels(rc.Spec.Template.Labels)
+			rc.Spec.Template.Labels["name"] = name
+			_, err = util.UpdateReplicationController(kubeClient, namespace, &rc)
+			if err != nil {
+				return fmt.Errorf("unable to update %s replication controller in namespace %s due to %+v", rc.GetName(), namespace, err)
+			}
+		}
+	}
+
+	services, err := util.ListServices(kubeClient, namespace, fmt.Sprintf("app=%s", appName))
+	if err != nil {
+		return fmt.Errorf("unable to list services for %s %s in namespace %s due to %+v", appName, name, namespace, err)
+	}
+
+	for _, svc := range services.Items {
+		if _, ok := svc.Labels["name"]; !ok {
+			svc.Labels = util.InitLabels(svc.Labels)
+			svc.Labels["name"] = name
+			_, err = util.UpdateService(kubeClient, namespace, &svc)
+			if err != nil {
+				return fmt.Errorf("unable to update %s service in namespace %s due to %+v", svc.GetName(), namespace, err)
+			}
+		}
+	}
+
+	configmaps, err := util.ListConfigMaps(kubeClient, namespace, fmt.Sprintf("app=%s", appName))
+	if err != nil {
+		return fmt.Errorf("unable to list config maps for %s %s in namespace %s due to %+v", appName, name, namespace, err)
+	}
+
+	for _, cm := range configmaps.Items {
+		if _, ok := cm.Labels["name"]; !ok {
+			cm.Labels = util.InitLabels(cm.Labels)
+			cm.Labels["name"] = name
```

Comment:
Should add "app" and "component" key/values to all these objects right?

Reply:
app and components are already exist for 2019.4.2. In order to support namespace scoped, just adding name.